### PR TITLE
Fix error handling

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/lumen/issues/585

Rather than having the LLM try to recover from unrecoverable errors, prevent it from trying to.

image

consolidates retries to prevent redundant code and only keeps the targeted retry while removing the catch-all-exceptions-and-retry since it usually fails to do the right thing and wastes tokens

image

### CODE REVIEW

File: lumen/ai/agents.py [hunk 1]
```diff
@@ -69,24 +69,17 @@ class Agent(Viewer):
...
-                    f"Taking a different approach to expertly resolve this issue `{exception}` using world-class knowledge.",
-                    user="Exception",
-                )
-            else:
-                self.interface.send(
-                    f"Error cannot be resolved: `{exception}`.",
-                    user="System",
-                    respond=False
-                )
+            if str(exception) in self.interface.serialize()[-1]["content"]:
```
Comment (ahuang11) [inline lumen/ai/agents.py:73]: Prevent redundant message sent twice.

File: lumen/ai/assistant.py [hunk 2]
```diff
@@ -296,6 +296,7 @@ async def _create_valid_agent(self, messages, system, agent_model, return_reason
             response_model=agent_model,
             allow_partial=False
         )
+
```
Comment (ahuang11) [inline lumen/ai/assistant.py:299]:
```suggestion
```

**Comment by droumis** [comment]
tested and works for me when missing API key

**Review by philippjfr — APPROVED** [review]

File: lumen/ai/agents.py (hunk 1 above)
Comment (philippjfr) [reply lumen/ai/agents.py:73]: Can you not just serialize the last message?

**Review by droumis — APPROVED** [review]

**Comment by ahuang11** [comment]
Planning to merge first and fix the serialize() once https://github.com/holoviz/panel/pull/6967 goes in.

**Pull request merged**